include/net/net_if.h

```diff
@@ -332,8 +352,10 @@ struct net_if_dev {
 	/** The hardware link address */
 	struct net_linkaddr link_addr;
 
-	/** Queue for outgoing packets from apps */
-	struct k_fifo tx_queue;
```

Comment:
you don't want to get rid of that, no. It should be either/or. 

On 1 tx queue (the common use-case currently) k_work_queue is going to bring an overhead as soon as you'll have 2+ ifaces. I actually centralized all in 1 unique tx thread last year exactly because of that: we were having 1 thread per-iface and that was really greedy on memory constrained system for 2+ ifaces being present. It's a regression on that aspect.

Reply:
Check net_if.c: net_if_tx_thread is unique. It runs for all net_if arounds (so all net devices).
k_work_queue here is going to break that (for instance if you have one net_if for bt ipsp, and one for 15.4.). 

Not sure how to tackle this however. Having ifdefs all around is not nice indeed.

Reply:
separating the traffic struct place holder as you did for ip configuration could be one solution.